types/kotlin2/UserProperties.kt

```diff
@@ -27,23 +27,23 @@ data class UserProperties(
         /**
          * Whether the user has the favourites space enabled
          */
-        val WebMetaSpaceFavouritesEnabled: Boolean? = null,
+        val webMetaSpaceFavouritesEnabled: Boolean? = null,
         /**
          * Whether the user has the home space set to all rooms
          */
-        val WebMetaSpaceHomeAllRooms: Boolean? = null,
+        val webMetaSpaceHomeAllRooms: Boolean? = null,
         /**
          * Whether the user has the home space enabled
          */
-        val WebMetaSpaceHomeEnabled: Boolean? = null,
+        val webMetaSpaceHomeEnabled: Boolean? = null,
         /**
          * Whether the user has the other rooms space enabled
          */
-        val WebMetaSpaceOrphansEnabled: Boolean? = null,
+        val webMetaSpaceOrphansEnabled: Boolean? = null,
         /**
          * Whether the user has the people space enabled
          */
-        val WebMetaSpacePeopleEnabled: Boolean? = null,
+        val webMetaSpacePeopleEnabled: Boolean? = null,
```

Comment:
This are the problems reported by Detekt, a static analysis tool.

Reply:
this looks like an issue in the source file https://github.com/matrix-org/matrix-analytics-events/blob/main/schemas/UserProperties.json

I'm not sure if posthog will handle casing as unique properties~

Reply:
That's why I did not change the value we send to Posthog but just the `internal` Kotlin code

Reply:
I think I’d agree with Adam that this should be fixed in the schema to maintain consistency.

Reply:
OK, did not want to break the existing schema. @novocaine OK for you (as it will impact the Web code)?

Reply:
Ah, because it’ll change the events sent to PostHog won’t it, yeah I’m guessing we probably don’t want to do that 🫤

Reply:
in this instance I believe the schema itself is inconsistent and would lean towards it being changed to avoid future confusing but I'm also wondering if the android project shouldn't be enforcing project conventions on the analytics repository? :thinking:

Reply:
I am happy to update the plan if confirmed by the Web maintainer.

Reply:
Can someone summarise the proposal here?

Reply:
We want to ensure all properties start with a lower case. Applicable to this one https://github.com/matrix-org/matrix-analytics-events/blob/d2ad5869fe479b7b1465b7b2f1f88178e9cc44c0/schemas/UserProperties.json#L23 and following.

Reply:
It'd probably be better to adjust the generator to skip things for Mobile which are prefixed with `Web` - rather than having to break existing posthog cohorts and lose historical data

Reply:
This PR adjust the Kotlin generator to generate valid Kotlin code without changing the data sent by Posthog. To me it was fine to do that. But I agree that mobile will not send `Web` property. This is something else which could be handled in another PR.

Would be happy to stop spending time on this little change which has *I think* no impact.

Reply:
The issue is we can't adjust the Typescript generator to tweak this without a lot of work given it is external to this project https://www.npmjs.com/package/json2ts

Reply:
I do not understand why it's preventing this PR to be merged. This PR does not change the plan and do not change the typescript code.

Reply:
I never said anything here is preventing this PR being merged.

I'm arguing with

> in this instance I believe the schema itself is inconsistent and would lean towards it being changed to avoid future confusion

Reply:
OK, sorry 🤗 . CC @ouchadam ^

Reply:
properties should be camelCase, I think that's fairly clear from examples in the README and in the code.

`Web` has been used as an enum prefix elsewhere which is why it was an uppercase W there, but I don't see why it should be so here, where we are describing a property. It should start with `web`. 

In terms of how disruptive this is to fix, I would be pretty surprised if posthog was case-sensitive..?